File: problems/908-smallest-range-i/solution.cpp
class Solution {
public:
    int smallestRangeI(vector<int>& nums, int k) {
        int minVal = nums[0];
        int maxVal = nums[0];
        for (int num : nums) {
            minVal = min(minVal, num);
            maxVal = max(maxVal, num);
        }
        int range = maxVal - minVal;
        return max(0, range - 2 * k);
    }
};

File: problems/908-smallest-range-i/solution.py
class Solution:
    def smallestRangeI(self, nums: List[int], k: int) -> int:
        min_num = min(nums)
        max_num = max(nums)
        diff = max_num - min_num
        if diff <= 2 * k:
            return 0
        else:
            return diff - 2 * k